gossipd/routing.c

```diff
@@ -877,17 +877,39 @@ static void dijkstra(struct routing_state *rstate,
 		     u64 riskbias,
 		     double fuzz, const struct siphash_seed *base_seed,
 		     struct unvisited *unvisited,
+		     size_t max_nodes,
+		     bool *nodes_limit,
```

Comment:
You're generally best-off expressing this as a "budget".  When it's zero, you stop.  Simplifies the logic somewhat.

Reply:
Good idea!